### PR TITLE
session: add delete_object

### PR DESCRIPTION
**Pull request by tarcieri**

### CODE REVIEW

**Comment by tarcieri** [comment]
No idea if this actually works... will go ahead and give it a try in another project with my branch

File: src/session.rs [hunk 1]
```diff
@@ -54,6 +54,19 @@ impl Session {
...
 
+    pub fn delete_object(&self, obj_id: u16, obj_type: ObjectType) -> Result<(), Error> {
+        unsafe {
+            match ReturnCode::from(yubihsm_sys::yh_util_delete_object(
+                self.this.get(),
+                obj_id,
+                obj_type.into()
+            )) {
+                ReturnCode::Success => Ok(()),
+                e => Err(format_err!("util_delete_object failed: {}", e)),
```
Comment (csssuf) [inline src/session.rs]: This should be `bail!` in keeping with #9.

File: src/session.rs [hunk 2]
```diff
@@ -54,6 +54,19 @@ impl Session {
         }
     }
 
+    pub fn delete_object(&self, obj_id: u16, obj_type: ObjectType) -> Result<(), Error> {
+        unsafe {
+            match ReturnCode::from(yubihsm_sys::yh_util_delete_object(
+                self.this.get(),
+                obj_id,
+                obj_type.into()
```
Comment (csssuf) [inline src/session.rs]: `rustfmt` makes this `obj_types.into(),`.

**Comment by tarcieri** [comment]
@csssuf updated

**Pull request merged**